Suppress virtual destructor and uncounted member variable warnings in WTF

rniwa opened the pull request:
#### e3948fcf9266c852214028357c1cad683ae06114
<pre>
Suppress virtual destructor and uncounted member variable warnings in WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=274434">https://bugs.webkit.org/show_bug.cgi?id=274434</a>

Reviewed by David Kilzer.

Suppress webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor in a few places.

This PR moves inline functions out of line for classes with virtual destructor warning suppressions
so that other checkers will continue to apply to their contents.

* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/JSONValues.h:
(WTF::JSONImpl::ObjectBase): Suppress the lack of virtual destructor warning.
(WTF::JSONImpl::ArrayBase): Ditto.
* Source/WTF/wtf/text/AtomStringImpl.cpp:
* Source/WTF/wtf/text/AtomStringImpl.h:
(WTF::AtomStringImpl): Ditto.
(WTF::AtomStringImpl::lookUp): Moved out of line so that other checkers will continue to apply to
function contents.
(WTF::AtomStringImpl::add): Ditto.
(WTF::AtomStringImpl::addCString): Ditto.
(WTF::AtomStringImpl::addWithStringTableProvider): Ditto.
* Source/WTF/wtf/text/StringConcatenate.h:
(WTF::StringTypeAdapter): Suppress the uncounted member variable warning.
* Source/WTF/wtf/text/StringView.cpp:
(WTF::StringView::UnderlyingString): Ditto.
* Source/WTF/wtf/text/SymbolImpl.h:
(WTF::SymbolImpl::SymbolImpl): Suppress the lack of virtual destructor warning.
(WTF::SymbolImpl::SymbolImpl): Moved out of line so that other checkers will continue to apply to
function contents.
(WTF::SymbolImpl::StaticSymbolImpl::StaticSymbolImpl): Ditto.
* Source/WTF/wtf/text/UniquedStringImpl.h:
(WTF::UniquedStringImpl): Suppress the lack of virtual destructor warning.
(WTF::UniquedStringImpl::UniquedStringImpl): Moved out of line so that other checkers will continue
to apply to function contents.

Canonical link: <a href="https://commits.webkit.org/279061@main">https://commits.webkit.org/279061@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/edc1e5793256a6e70c1877eb392b1dbd1c7134ab

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52318 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31650 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4739 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55592 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3041 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38074 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2740 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42550 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1944 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54414 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29279 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45155 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23627 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26511 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2430 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1200 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/45667 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48412 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2580 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57188 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/51826 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27444 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2613 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49947 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28677 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45275 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49199 "Found 2 new API test failures: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure, /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/proxy (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29589 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/64133 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7676 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28422 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12138 "Found 254 new JSC stress test failures: basic-tests.yaml/stress-test.js.bytecode-cache, basic-tests.yaml/stress-test.js.dfg-eager, basic-tests.yaml/stress-test.js.dfg-eager-no-cjit-validate, basic-tests.yaml/stress-test.js.eager-jettison-no-cjit, basic-tests.yaml/stress-test.js.no-cjit-collect-continuously, basic-tests.yaml/stress-test.js.no-cjit-validate-phases, jsc-layout-tests.yaml/js/script-tests/statement-list-item-syntax-errors.js.layout-dfg-eager-no-cjit, microbenchmarks/large-map-iteration.js.bytecode-cache, stress/array-flatmap.js.dfg-eager, stress/big-int-unary-plus.js.dfg-eager-no-cjit-validate ... (failure)") | 
<!--EWS-Status-Bubble-End-->

rniwa committed:
Logger should have a virtual destructor
https://bugs.webkit.org/show_bug.cgi?id=274432

Reviewed by David Kilzer and Chris Dumez.

Added the virtual destructor since AggregateLogger subclasses Logger
and has non-POD member variables.

* Source/WTF/wtf/Logger.h:
(WTF::Logger::~Logger):

Canonical link: https://commits.webkit.org/279033@main

mwyrzykowski committed:
[WebGPU] Render / compute commands should be no-ops when the command buffer will be discarded
https://bugs.webkit.org/show_bug.cgi?id=274290
<radar://128202440>

Reviewed by Dan Glastonbury.

There was validation logic however it did not take into account
destroyed render targets, which are still valid to call commands
on.

A destroyed render target will result in the command buffer being
discarded when it is submitted, so we can make all the encoding
operations no-ops.

* LayoutTests/TestExpectations:
* LayoutTests/fast/webgpu/fuzz-274290-expected.txt: Added.
* LayoutTests/fast/webgpu/fuzz-274290.html: Added.
Add regression test.

* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::submitWillBeInvalid const):
* Source/WebGPU/WebGPU/ComputePassEncoder.h:
(WebGPU::ComputePassEncoder::isValid const): Deleted.
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::executePreDispatchCommands):
(WebGPU::ComputePassEncoder::dispatch):
(WebGPU::ComputePassEncoder::runPredispatchIndirectCallValidation):
(WebGPU::ComputePassEncoder::dispatchIndirect):
(WebGPU::ComputePassEncoder::setBindGroup):
(WebGPU::ComputePassEncoder::isValid const):
(WebGPU::ComputePassEncoder::computeCommandEncoder const):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::beginOcclusionQuery):
(WebGPU::RenderPassEncoder::endOcclusionQuery):
(WebGPU::RenderPassEncoder::errorValidatingAndBindingBuffers):
(WebGPU::RenderPassEncoder::executePreDrawCommands):
(WebGPU::RenderPassEncoder::draw):
(WebGPU::RenderPassEncoder::drawIndexed):
(WebGPU::RenderPassEncoder::drawIndexedIndirect):
(WebGPU::RenderPassEncoder::drawIndirect):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::renderCommandEncoder const):
(WebGPU::RenderPassEncoder::setBindGroup):
(WebGPU::RenderPassEncoder::setBlendConstant):
(WebGPU::RenderPassEncoder::setScissorRect):
(WebGPU::RenderPassEncoder::setStencilReference):
Call helper function so command recording becomes no-ops for
command buffers which will be discarded.

Canonical link: https://commits.webkit.org/279034@main

cdumez committed:
PublicSuffixStore::platformTopPrivatelyControlledDomain() should take in a StringView
https://bugs.webkit.org/show_bug.cgi?id=274433

Reviewed by Ryosuke Niwa.

PublicSuffixStore::platformTopPrivatelyControlledDomain() should take in a StringView
instead of a String. It keeps calling substring(), which is a lot more efficient on a
StringView than a String.

* Source/WebCore/platform/PublicSuffixStore.h:
* Source/WebCore/platform/cocoa/PublicSuffixStoreCocoa.mm:
(WebCore::PublicSuffixStore::platformTopPrivatelyControlledDomain const):
* Source/WebCore/platform/network/curl/PublicSuffixStoreCurl.cpp:
(WebCore::PublicSuffixStore::platformTopPrivatelyControlledDomain const):
* Source/WebCore/platform/soup/PublicSuffixStoreSoup.cpp:
(WebCore::permissiveTopPrivateDomain):
(WebCore::PublicSuffixStore::platformTopPrivatelyControlledDomain const):

Canonical link: https://commits.webkit.org/279035@main

annevk committed:
Implement Request & Response's bytes() method
https://bugs.webkit.org/show_bug.cgi?id=274102

Reviewed by Youenn Fablet.

This accessor is nearly identical to arrayBuffer() except that it
yields a Uint8Array, which is strictly more useful.

Fetch change: https://github.com/whatwg/fetch/pull/1753.

* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/general.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/general.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/general.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/general.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/abort/serviceworker-intercepted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/request/request-consume.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/request/request-consume.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/request/request-consume.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/request/request-consume.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-blob-realm.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-error-from-stream.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-error-from-stream.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-error-from-stream.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-error-from-stream.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-stream-bad-chunk.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-stream-bad-chunk.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-stream-bad-chunk.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/response/response-stream-bad-chunk.any.worker-expected.txt:
* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::bytes):
* Source/WebCore/Modules/fetch/FetchBody.h:
* Source/WebCore/Modules/fetch/FetchBody.idl:
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::resolveWithTypeAndData):
(WebCore::FetchBodyConsumer::resolve):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.h:
* Source/WebCore/Modules/fetch/FetchBodyOwner.cpp:
(WebCore::FetchBodyOwner::bytes):
* Source/WebCore/Modules/fetch/FetchBodyOwner.h:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
(WebCore::fulfillPromiseWithUint8Array):
(WebCore::fulfillPromiseWithUint8ArrayFromSpan):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:

Canonical link: https://commits.webkit.org/279036@main

Constellation committed:
GlyphDisplayListCache should follow to traditional cache protocol
https://bugs.webkit.org/show_bug.cgi?id=274380
rdar://128376419

Reviewed by Alan Baradlay.

We found that GlyphDisplayListCache is not effective much. So much time is used by Glyph DisplayList creation.
In this patch, we change the cache to 2-tier-based.

1. If layer is frequently painted, then do caching anyway. It is aligned to the current behavior.
2. Still we have a front cache, which has limited size. And stop doing a cache when it reaches to the threshold.
   Since cache entries are wiped when LayoutRun gets destroyed, size of Cache is not monotonically increasing.
   Rather it stays within small size.

* LayoutTests/TestExpectations: We found that these tests are already broken before this patch (if you change the window size to flush, then it disappears). So listing them.
* Source/WebCore/rendering/GlyphDisplayListCache.cpp:
(WebCore::GlyphDisplayListCache::clear):
(WebCore::GlyphDisplayListCache::getDisplayList):
(WebCore::GlyphDisplayListCache::get):
(WebCore::GlyphDisplayListCache::getIfExists):
(WebCore::GlyphDisplayListCache::remove):
* Source/WebCore/rendering/GlyphDisplayListCache.h:
* Source/WebCore/rendering/TextPainter.cpp:
(WebCore::TextPainter::shouldUseGlyphDisplayList):
* Source/WebCore/rendering/TextPainter.h:
(WebCore::TextPainter::setGlyphDisplayListIfNeeded):

Canonical link: https://commits.webkit.org/279037@main

dpino committed:
[LibWebRTC][GTK] Unreviewed, fix build after 278972@main
https://bugs.webkit.org/show_bug.cgi?id=274297

* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoEncoderFactory.cpp:
(WebCore::GStreamerVideoEncoder::makeElement):

Canonical link: https://commits.webkit.org/279038@main

Commit:
`m_maxMessageSize` should be `infinity()` instead of `max()` in `RTCSctpTransport.h`

https://bugs.webkit.org/show_bug.cgi?id=274025
rdar://problem/128306030

Reviewed by Youenn Fablet.

This patch aligns WebKit with web specification [1]:

[1] https://w3c.github.io/webrtc-pc/#dfn-update-the-data-max-message-size

"If both remoteMaxMessageSize and canSendSize are 0, set
[[MaxMessageSize]] to the positive Infinity value."

* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
(double m_maxMessageSize):
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCSctpTransport-maxMessageSize-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/279039@main

obyknovenius committed:
[GTK][WPE]: Regression (278519@main): build error with gcc 12.3.0
https://bugs.webkit.org/show_bug.cgi?id=274072

Reviewed by Sam Weinig.

To parse a CSS time value in `CSSPropertyParserHelpers::consumeTime()`,
we call `CSSPrimitiveValueResolver<TimeRaw>::consumeAndResolve()` static
method.

This method consumes a range of tokens, creates a `MetaConsumeResult`,
which is essentially a `std::variant<TimeRow, UnevaluatedCalc<TimeRaw>`,
and passes it to `CSSPrimitiveValueResolver<TimeRaw>::resolve()`.

Then `WTF::switchOn()` is used to get the `consumeResult` value.
Beacuse `UnevaluatedCalc` is a struct that includes
a `Ref<CSSCalcValue>` `calc`, and it is passed by value to
`WTF::switchOn()` lambda, GCC believes that it's there is a possibility
of the `calc` value being destroyed but still reachable.

Passing `consumeResult` value by reference fixes the warning and reduces
reference-counting churn.

* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:
(WebCore::CSSPropertyParserHelpers::MetaResolver::resolve):

Canonical link: https://commits.webkit.org/279040@main

cdumez committed:
PublicSuffixStore::m_canAcceptCustomPublicSuffix doesn't need synchronization
https://bugs.webkit.org/show_bug.cgi?id=274431

Reviewed by Darin Adler and Ryosuke Niwa.

PublicSuffixStore::m_canAcceptCustomPublicSuffix doesn't need synchronization
as it is only ever used on the main thread.

* Source/WebCore/platform/PublicSuffixStore.h:
(WebCore::PublicSuffixStore::WTF_GUARDED_BY_LOCK): Deleted.
* Source/WebCore/platform/cocoa/PublicSuffixStoreCocoa.mm:
(WebCore::PublicSuffixStore::enablePublicSuffixCache):
(WebCore::PublicSuffixStore::addPublicSuffix):

Canonical link: https://commits.webkit.org/279041@main

rwlbuis committed:
Adopt more smart pointers in rendering/mathml
https://bugs.webkit.org/show_bug.cgi?id=274384

Reviewed by Sihui Liu.

Adopt more smart pointers in endering/mathml based on the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/rendering/mathml/RenderMathMLBlock.cpp:
(WebCore::axisHeight):
* Source/WebCore/rendering/mathml/RenderMathMLFenced.cpp:
(WebCore::RenderMathMLFenced::updateFromElement):
* Source/WebCore/rendering/mathml/RenderMathMLFraction.cpp:
(WebCore::RenderMathMLFraction::defaultLineThickness const):
(WebCore::RenderMathMLFraction::fractionParameters const):
(WebCore::RenderMathMLFraction::stackParameters const):
* Source/WebCore/rendering/mathml/RenderMathMLRoot.cpp:
(WebCore::RenderMathMLRoot::horizontalParameters):
(WebCore::RenderMathMLRoot::verticalParameters):
* Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp:
(WebCore::RenderMathMLScripts::spaceAfterScript):
(WebCore::RenderMathMLScripts::verticalParameters const):
* Source/WebCore/rendering/mathml/RenderMathMLSpace.cpp:
(WebCore::RenderMathMLSpace::spaceWidth const):
(WebCore::RenderMathMLSpace::getSpaceHeightAndDepth const):
* Source/WebCore/rendering/mathml/RenderMathMLToken.cpp:
(WebCore::RenderMathMLToken::updateMathVariantGlyph):
* Source/WebCore/rendering/mathml/RenderMathMLUnderOver.cpp:
(WebCore::RenderMathMLUnderOver::verticalParameters const):

Canonical link: https://commits.webkit.org/279042@main

Commit:
Sync 'css/geometry' tests from WPT upstream

https://bugs.webkit.org/show_bug.cgi?id=265211
rdar://problem/118893090

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/13d894d575c4ad428b8149453bcb4ebf9f11ead6

* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix2DInit-validate-fixup-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix2DInit-validate-fixup.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMRectList-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMRectList.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.worker-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.worker.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/support/dommatrix-test-util.js:
(checkMatrix):
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/support/interfaces.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/support/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.any.worker.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.any.js:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.any.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix-invert-preserves-2d.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix-invert-preserves-2d-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix-invert-non-invertible.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix-invert-non-invertible-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix-invert-invertible.html:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/DOMMatrix-invert-invertible-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/geometry/idlharness.worker.html: Removed.

Canonical link: https://commits.webkit.org/279043@main

rkirsling committed:
Partially fix PS build after 278995@main
https://bugs.webkit.org/show_bug.cgi?id=274455

Unreviewed build fix.

The majority of this build fix needs to happen outside of WebKit, but either way there's a missing include.

* Source/WebKit/Platform/CoroutineUtilities.h:

Canonical link: https://commits.webkit.org/279044@main

ziransun committed:
Improve test coverage of sink values for trusted types [part1].
https://bugs.webkit.org/show_bug.cgi?id=273763

Reviewed by Manuel Rego Casasnovas.

Check the sink value for defaul policy in the tests.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/block-Document-execCommand.html:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/block-string-assignment-to-DOMWindowTimers-setTimeout-setInterval.html:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/modify-attributes-in-callback.html:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/require-trusted-types-for.html:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-createHTMLDocument.html:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-duplicate-names-without-enforcement.html:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/worker-constructor.https.html:

Canonical link: https://commits.webkit.org/279045@main

dpino committed:
[GCC] Unreviewed, build fix for Debian 11 after 278995@main
https://bugs.webkit.org/show_bug.cgi?id=267827

GCC10 requires compiler flag '-fcoroutines'.

* Source/cmake/WebKitCompilerFlags.cmake:

Canonical link: https://commits.webkit.org/279046@main

Constellation committed:
[JSC] Split super complicated IC code generation clause into three different ones
https://bugs.webkit.org/show_bug.cgi?id=274429
rdar://128435898

Reviewed by Justin Michaud.

Load, GetGetter, CustomValueGetter, CustomValueSetter, CustomAccessorGetter, CustomAccessorSetter, Getter, Setter, and IntrinsicGetter are
sharing one switch-clause, and it makes code incredibly complicated and hard to follow. Let's just split them into three.

1. Load, GetGetter
2. CustomValueGetter, CustomValueSetter, CustomAccessorGetter, CustomAccessorSetter
3. Getter, Setter, IntrinsicGetter

This significantly reduces complexity for simple IC (like Load), and it also improves some code generation for CustomAccessorGetter / CustomAccessorSetter,
where we are unnecessarily loading property owner GPR before.

* Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp:
(JSC::InlineCacheCompiler::generateImpl):

Canonical link: https://commits.webkit.org/279047@main

Constellation committed:
[JSC] Ensure using CLoop for x86 (32bit)
https://bugs.webkit.org/show_bug.cgi?id=274452
rdar://128449662

Reviewed by Ross Kirsling.

This patch ensures that CLoop is enabled on x86 (32bit) and dropping asm LLInt support.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/llint/LLIntOfflineAsmConfig.h:
* Source/JavaScriptCore/llint/LowLevelInterpreter.asm:
* Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm:
* Source/JavaScriptCore/offlineasm/backends.rb:
* Source/JavaScriptCore/offlineasm/x86.rb:
* Source/cmake/WebKitFeatures.cmake:

Canonical link: https://commits.webkit.org/279048@main

mattwoodrow committed:
[view-transitions] https://pokedex-dev.vercel.app/pokemons has super low framerate.
https://bugs.webkit.org/show_bug.cgi?id=274439
<rdar://128030956>

Reviewed by Tim Nguyen.

We invalidate the stylesheet every frame, resulting in way too much painting.

Instead, only invalidate the stylesheet if we actually inserted new styles.
Also invalidate layout, and the layer configuration for the pseudo if we
change those.

* Source/WebCore/css/MutableStyleProperties.cpp:
(WebCore::MutableStyleProperties::mergeAndOverrideOnConflict):
* Source/WebCore/css/MutableStyleProperties.h:
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::ViewTransition::updatePseudoElementStyles):
* Source/WebCore/rendering/RenderViewTransitionCapture.cpp:
(WebCore::RenderViewTransitionCapture::setSize):
* Source/WebCore/rendering/RenderViewTransitionCapture.h:

Canonical link: https://commits.webkit.org/279049@main

Constellation committed:
Unreviewed, follow-up after 279021@main
https://bugs.webkit.org/show_bug.cgi?id=274445
rdar://128405738

* Source/JavaScriptCore/bytecode/AccessCase.cpp:
(JSC::AccessCase::canBeShared):
* Source/JavaScriptCore/bytecode/SharedJITStubSet.h:
(JSC::SharedJITStubSet::Searcher::Translator::hash):
(JSC::SharedJITStubSet::Searcher::Translator::equal):
(JSC::SharedJITStubSet::Searcher::Searcher):

Canonical link: https://commits.webkit.org/279050@main

Constellation committed:
Continue cleaning up compiler configurations
https://bugs.webkit.org/show_bug.cgi?id=274438
rdar://128441724

Reviewed by Ross Kirsling.

This patch further simplifies compiler configurations.

1. Drop MINGW, nobody is supporting it.
2. Remove MSVC hacks further.
3. Remove most of COMPILER(GCC_COMPATIBLE) in JavaScriptCore.
4. Fix CodeBlock's static_assert, which was not effective before.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/ARM64Assembler.h:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerARMv7.cpp:
* Source/JavaScriptCore/bytecode/CodeBlock.h:
* Source/JavaScriptCore/heap/GCMemoryOperations.h:
(JSC::gcSafeMemcpy):
(JSC::gcSafeMemmove):
(JSC::gcSafeZeroMemory):
* Source/JavaScriptCore/jit/CCallHelpers.h:
(JSC::CCallHelpers::marshallArgumentRegister):
* Source/JavaScriptCore/jit/JITOperations.cpp:
(JSC::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/runtime/ArrayConventions.cpp:
(JSC::clearArrayMemset):
* Source/JavaScriptCore/runtime/EnsureStillAliveHere.cpp: Removed.
* Source/JavaScriptCore/runtime/EnsureStillAliveHere.h:
* Source/JavaScriptCore/runtime/ExceptionScope.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSBigInt.cpp:
(JSC::JSBigInt::digitDiv):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/CurrentTime.cpp:
(WTF::highResUpTime):
* Source/WTF/wtf/InlineASM.h:
* Source/WTF/wtf/win/ThreadingWin.cpp:
(WTF::Thread::initializeCurrentThreadInternal):
* Source/WebCore/platform/graphics/FormatConverter.cpp:
(WebCore::FormatConverter::convert):

Canonical link: https://commits.webkit.org/279051@main

philn committed:
[GStreamer] GstVideoFrame leaks
https://bugs.webkit.org/show_bug.cgi?id=274257

Reviewed by Xabier Rodriguez-Calvar.

The ImageGStreamer no longer holds a BitmapImage, but a PlatformImagePtr. A BitmapImage is now
created by VideoFrameGStreamer when painting is required. The ImageGStreamerSkia implementation no
longer holds the mapped GstVideoFrame because that keeps un-necessary references and file
descriptors open, the needed plane data is copied instead. The static ImageDecoderGStreamer vector
now stores WeakPtrs instead of RefPtr, which semantically more correct but in practice makes no difference.

* Source/WebCore/platform/graphics/gstreamer/ImageDecoderGStreamer.cpp:
(WebCore::ensureDebugCategoryIsInitialized):
(WebCore::teardownGStreamerImageDecoders):
(WebCore::ImageDecoderGStreamer::create):
(WebCore::ImageDecoderGStreamer::ImageDecoderGStreamer):
* Source/WebCore/platform/graphics/gstreamer/ImageDecoderGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/ImageGStreamer.h:
(WebCore::ImageGStreamer::create):
(WebCore::ImageGStreamer::image const):
(WebCore::ImageGStreamer::rect):
(WebCore::ImageGStreamer::createImage): Deleted.
(WebCore::ImageGStreamer::image): Deleted.
* Source/WebCore/platform/graphics/gstreamer/ImageGStreamerCairo.cpp:
(WebCore::ImageGStreamer::ImageGStreamer):
* Source/WebCore/platform/graphics/gstreamer/ImageGStreamerSkia.cpp:
(WebCore::ImageGStreamer::ImageGStreamer):
* Source/WebCore/platform/graphics/gstreamer/MediaSampleGStreamer.h:
(WebCore::MediaSampleGStreamer::sample const):
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::convertSampleToImage):
(WebCore::VideoFrame::paintInContext):

Canonical link: https://commits.webkit.org/279052@main

tadeuzagallo committed:
[WGSL] Type checker does not handle shadowing of overloaded built-ins correctly
https://bugs.webkit.org/show_bug.cgi?id=274391
rdar://128202008

Reviewed by Mike Wyrzykowski.

The type checker incorrectly looked up the built-in definition even there was another
definition shadowing it.

* Source/WebGPU/WGSL/TypeCheck.cpp:
(WGSL::TypeChecker::visit):
* Source/WebGPU/WGSL/tests/invalid/function-call.wgsl:

Canonical link: https://commits.webkit.org/279053@main

GeorgesStavracas committed:
[GTK] Undeprecate and document webkit_print_operation_print() behavior
https://bugs.webkit.org/show_bug.cgi?id=274456

Reviewed by Carlos Garcia Campos.

Turns out, there might be legitimate use cases for webkit_print_operation_print()
for sandboxed apps. However, they're very specific, and this function only works
in very specific situations.

Document that, and undeprecate the function.

* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.h.in:

Canonical link: https://commits.webkit.org/279054@main

justinmichaud committed:
Refactor OMG call patchpoint and tail call patchpoint
https://bugs.webkit.org/show_bug.cgi?id=273823

Reviewed by Yusuke Suzuki.

The wasm tail calls feature does not currently work when the new frame
overlaps the old frame in interesting ways, so the tail calls tests
were disabled. Then, OMG inlining broke tail calls, causing an assertion failure.

This patch does not fix tail calls, but it does fix that assertion.

It also refactors both versions of create[Tail]CallPatchpoint to look
the same. This makes it a little nicer to read, but more importantly,
it makes it easier for a follow-up patch to fix OMG tail calls.

The main reason for this change is so that the follow-up patch is easier to read.

* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::emitIndirectCall):
(JSC::Wasm::OMGIRGenerator::createCallPatchpoint):
(JSC::Wasm::OMGIRGenerator::createTailCallPatchpoint):
(JSC::Wasm::OMGIRGenerator::addCall):

Canonical link: https://commits.webkit.org/279055@main

sosukesuzuki committed:
Add Sosuke Suzuki to contributors.json
https://bugs.webkit.org/show_bug.cgi?id=274466

Reviewed by Jonathan Bedard.

* metadata/contributors.json:

Canonical link: https://commits.webkit.org/279056@main

Commit:
Add an OOME check in ClonedArguments::createEmpty.
https://bugs.webkit.org/show_bug.cgi?id=274451
rdar://121951051

Reviewed by Yusuke Suzuki.

Also added a RELEASE_ASSERT in operationCreateClonedArgumentsDuringExit() to make it explicit
that we're not allowed to have OOMEs (from ClonedArguments::createEmpty() or othewise) during
an OSR exit.  Encountering an OOME then is considered a non-recoverable event.

* Source/JavaScriptCore/dfg/DFGOperations.cpp:
(JSC::DFG::JSC_DEFINE_JIT_OPERATION):
* Source/JavaScriptCore/ftl/FTLOperations.cpp:
(JSC::FTL::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/runtime/ClonedArguments.cpp:
(JSC::ClonedArguments::createEmpty):
(JSC::ClonedArguments::createWithInlineFrame):
(JSC::ClonedArguments::createByCopyingFrom):
* Source/JavaScriptCore/runtime/ClonedArguments.h:
* Source/JavaScriptCore/runtime/JSObject.cpp:
(JSC::createArrayStorageButterflyImpl):
(JSC::JSObject::createArrayStorageButterfly):
(JSC::JSObject::tryCreateArrayStorageButterfly):
* Source/JavaScriptCore/runtime/JSObject.h:

Canonical link: https://commits.webkit.org/279057@main

ddkilzer reviewed (approved):
r=me

Do we need to leave `ALWAYS_INLINE` on the declarations, or is it sufficient to add that only to the method implementations after the class?

webkit-commit-queue committed:
Unreviewed, reverting 279031@main.
https://bugs.webkit.org/show_bug.cgi?id=274472

SP3 and JS2 regressions

Reverted changeset:

"exceptions can't always be checked under silent spill in DFG"
https://bugs.webkit.org/show_bug.cgi?id=274291
https://commits.webkit.org/279031@main

Canonical link: https://commits.webkit.org/279058@main

bnham committed:
Avoid duplicate ReleaseMemory calls on Mac
https://bugs.webkit.org/show_bug.cgi?id=274396
rdar://127750149

Reviewed by Chris Dumez.

On the Mac, we enable NON_VISIBLE_WEBPROCESS_MEMORY_CLEANUP_TIMER, which calls releaseMemory to
clean up memory two minutes after a WebContent process has no visible webpages. We also call
releaseMemory just before a process suspends (in `WebProcessProxy::prepareToDropLastAssertion`),
whcih generally happens 8 minutes after a WebContent process has no visible webpages (see
`_wasRecentlyVisibleActivity` in WebPageProxy).

This means we are calling ReleaseMemory twice as a WebContent process transitions from foreground to
background to suspended states. We only meant to call this once. Fix this by disabling the call to
releaseMemory in prepareToDropLastAssertion if the non-visible cleanup timer is enabled.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::prepareToDropLastAssertion):

Canonical link: https://commits.webkit.org/279059@main

lukewarlow committed:
Import close-watcher WPT tests
https://bugs.webkit.org/show_bug.cgi?id=266379

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/43ae6cd615fad8cd6df2cade33ad48bfb9f285cb

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/abortsignal-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/abortsignal.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/basic-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/basic.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/keydown-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/keydown.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/keypress-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/keypress.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/keyup-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/keyup.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/not-user-activation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/not-user-activation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/synthetic-keyboard-event-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/synthetic-keyboard-event.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/esc-key/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/event-properties-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/event-properties.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/frame-removal-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/frame-removal.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/inside-event-listeners-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/inside-event-listeners.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/resources/helpers.js: Added.
(window.createRecordingCloseWatcher):
(window.createBlessedRecordingCloseWatcher.async t):
(window.destroyCloseWatcher):
(window.sendEscKey):
(window.maybeTopLayerBless):
(window.waitForPotentialCloseEvent):
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-activate-preventDefault.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-activate-preventDefault_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-activate-preventDefault_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-activate.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-activate_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-activate_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-closerequest-n.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-closerequest-n_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-closerequest-n_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-destroy-n.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-destroy-n_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n-destroy-n_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/n_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-CloseWatcher.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-activate-CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-activate-CloseWatcher.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-activate-dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-activate-dialog.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nn-dialog.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-CloseWatcher-dialog-popover-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-CloseWatcher-dialog-popover.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-CloseWatcher.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-dialog.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-popovers-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nnn-popovers.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ny-activate-preventDefault.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ny-activate-preventDefault_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ny-activate-preventDefault_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ny.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ny_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ny_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyn-popovers-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyn-popovers.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyn.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyn_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyn_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nynn-destroy.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nynn-destroy_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nynn-destroy_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nynn.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nynn_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nynn_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyn-CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyn-CloseWatcher.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyn-dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyn-dialog.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyyn-CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyyn-CloseWatcher.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyyn-dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/nyyyn-dialog.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/y.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/y_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/y_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yn-activate.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yn-activate_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yn-activate_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yn.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yn_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yn_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ynn-CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ynn-CloseWatcher.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ynn-dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/ynn-dialog.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yy.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yy_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yy_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyn.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyn_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyn_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy-CloseWatcher-dialog-popover-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy-CloseWatcher-dialog-popover.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy-activate-CloseWatcher-dialog-popover-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy-activate-CloseWatcher-dialog-popover.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy-popovers-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy-popovers.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy_CloseWatcher-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/user-activation/yyy_dialog-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/close-watcher/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-close-request-expected.txt:
* LayoutTests/TestExpectations:

Canonical link: https://commits.webkit.org/279060@main

rniwa committed:
Suppress virtual destructor and uncounted member variable warnings in WTF
https://bugs.webkit.org/show_bug.cgi?id=274434

Reviewed by David Kilzer.

Suppress webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor in a few places.

This PR moves inline functions out of line for classes with virtual destructor warning suppressions
so that other checkers will continue to apply to their contents.

* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/JSONValues.h:
(WTF::JSONImpl::ObjectBase): Suppress the lack of virtual destructor warning.
(WTF::JSONImpl::ArrayBase): Ditto.
* Source/WTF/wtf/text/AtomStringImpl.cpp:
* Source/WTF/wtf/text/AtomStringImpl.h:
(WTF::AtomStringImpl): Ditto.
(WTF::AtomStringImpl::lookUp): Moved out of line so that other checkers will continue to apply to
function contents.
(WTF::AtomStringImpl::add): Ditto.
(WTF::AtomStringImpl::addCString): Ditto.
(WTF::AtomStringImpl::addWithStringTableProvider): Ditto.
* Source/WTF/wtf/text/StringConcatenate.h:
(WTF::StringTypeAdapter): Suppress the uncounted member variable warning.
* Source/WTF/wtf/text/StringView.cpp:
(WTF::StringView::UnderlyingString): Ditto.
* Source/WTF/wtf/text/SymbolImpl.h:
(WTF::SymbolImpl::SymbolImpl): Suppress the lack of virtual destructor warning.
(WTF::SymbolImpl::SymbolImpl): Moved out of line so that other checkers will continue to apply to
function contents.
(WTF::SymbolImpl::StaticSymbolImpl::StaticSymbolImpl): Ditto.
* Source/WTF/wtf/text/UniquedStringImpl.h:
(WTF::UniquedStringImpl): Suppress the lack of virtual destructor warning.
(WTF::UniquedStringImpl::UniquedStringImpl): Moved out of line so that other checkers will continue
to apply to function contents.

Canonical link: https://commits.webkit.org/279061@main
